pkg/envconfig/envconfig.go

```diff
@@ -0,0 +1,12 @@
+package envconfig
+
+type EnvConfig struct {
+	Port                       int    `envconfig:"PORT" required:"true"`
+	Domain                     string `envconfig:"STS_DOMAIN" required:"true"`
+	KMSKey                     string `envconfig:"KMS_KEY" required:"false"`
+	AppID                      int64  `envconfig:"GITHUB_APP_ID" required:"true"`
+	EventingIngress            string `envconfig:"EVENT_INGRESS_URI" required:"true"`
+	AppSecretCertificateFile   string `envconfig:"APP_SECRET_CERTIFICATE_FILE" required:"false"`
+	AppSecretCertificateEnvVar string `envconfig:"APP_SECRET_CERTIFICATE_ENV_VAR" required:"false"`
+	Metrics                    bool   `envconfig:"METRICS" required:"false" default:"true"`
+}
```

Comment:
It's probably worth adding some validation to return an error if multiple KMS options are set.

e.g. 

```go
func Process() (*EnvConfig, error) {
  cfg := new(EnvConfig)
  if err := envconfig.Process("", cfg); err != nil {
    return nil, err
  }

  kmsSet := false
  for _, v := range []string{cfg.KMSKey, cfg,AppSecretCertificateFile, cfg.AppSecretCertificateEnvVar} {
    if v != "" {
      if kmsSet {
        return nil, errors.New("only one of KMS_KEY, APP_SECRET_CERTIFICATE_FILE, APP_SECRET_CERTIFICATE_ENV_VAR may be set")
      }
      kmsSet = true
    }
  }
  
  return cfg, err

Reply:
[878994d](https://github.com/octo-sts/app/pull/412/commits/878994d0e542db889818fb31943927dcbd2f8c29) should do it. thanks